compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -977,12 +977,22 @@ impl<'a> Parser<'a> {
 
     fn parse_dot_or_call_expr_with_(&mut self, mut e: P<Expr>, lo: Span) -> PResult<'a, P<Expr>> {
         loop {
-            if self.eat(&token::Question) {
+            let has_question = if self.prev_token.kind == TokenKind::Ident(kw::Return, false) {
+                self.eat_noexpect(&token::Question)
```

Comment:
Can you leave a comment why this is noexpect? And same for below.

Reply:
Done.